torchvision/prototype/datasets/_builtin/semeion.py

```diff
@@ -0,0 +1,69 @@
+import io
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+import torch
+from torchdata.datapipes.iter import (
+    IterDataPipe,
+    Mapper,
+    Shuffler,
+    CSVParser,
+)
+from torchvision.prototype.datasets.decoder import raw
+from torchvision.prototype.datasets.utils import (
+    Dataset,
+    DatasetConfig,
+    DatasetInfo,
+    HttpResource,
+    OnlineResource,
+    DatasetType,
+)
+from torchvision.prototype.datasets.utils._internal import INFINITE_BUFFER_SIZE, image_buffer_from_array
+
+
+class SEMEION(Dataset):
+    def _make_info(self) -> DatasetInfo:
+        return DatasetInfo(
+            "semeion",
+            type=DatasetType.RAW,
+            categories=10,
+            homepage="https://archive.ics.uci.edu/ml/datasets/Semeion+Handwritten+Digit",
+        )
+
+    def resources(self, config: DatasetConfig) -> List[OnlineResource]:
+        archive = HttpResource(
+            "http://archive.ics.uci.edu/ml/machine-learning-databases/semeion/semeion.data",
+            sha256="f43228ae3da5ea6a3c95069d53450b86166770e3b719dcc333182128fe08d4b1",
+        )
+        return [archive]
+
+    def _collate_and_decode_sample(
+        self,
+        data: Tuple[str, ...],
+        *,
+        decoder: Optional[Callable[[io.IOBase], torch.Tensor]],
+    ) -> Dict[str, Any]:
+        image_data = torch.tensor([float(pixel) for pixel in data[:256]]).reshape(16, 16)
```

Comment:
```suggestion
        image_data = torch.tensor([int(pixel) for pixel in data[:256]], dtype=torch.uint8).reshape(16, 16)
```

Reply:
Thanks, @pmeier - but I think this wouldn't work:

```python
ValueError: invalid literal for int() with base 10: '0.0000'
```

However, I feel passing `dtype=torch.uint8` is better than doing `.to(torch.uint8)`, so I'll edit that. :)

Reply:
Done! :)